### PR TITLE
Implement token parsing with custom key

### PR DESCRIPTION
**Pull request by GerritSe**

### CODE REVIEW

File: compeon-rack_tools.gemspec [hunk 1]
```diff
@@ -22,6 +22,7 @@ Gem::Specification.new do |spec|
   end
   spec.require_paths = ['lib']
 
+  spec.add_dependency 'compeon-access_token'
```
Comment (timoschilling) [inline compeon-rack_tools.gemspec]: I still think it should be an optional dependency!

File: lib/compeon/rack_tools/pipes/parse_token_with_key.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+require 'compeon/rack_tools/http_errors'
+require 'compeon/token'
+
+module Compeon
+  module RackTools
+    module Pipes
+      PARSE_TOKEN_WITH_KEY = lambda do |key:|
+        lambda do |request:, **rest|
+          token_string = request.env['HTTP_AUTHORIZATION']&.match(/^token (?<token>.*)/)&.named_captures&.[]('token')
```
Comment (lagr) [inline lib/compeon/rack_tools/pipes/parse_token_with_key.rb]: I think this is much more readable, would you agree? (It is not so clever, though)
```suggestion
          auth_header = request.env.fetch('HTTP_AUTHORIZATION', '')
          token_string = auth_header[/^token (.*)/, 1]
```

Comment (GerritSe) [reply lib/compeon/rack_tools/pipes/parse_token_with_key.rb]: Actually it's a copy pasta from `PARSE_TOKEN` and I just replaced what was necessary :D

Comment (lagr) [reply lib/compeon/rack_tools/pipes/parse_token_with_key.rb]: Or, if you insist on keeping it a one-liner
```ruby
token_string = request.env.fetch('HTTP_AUTHORIZATION', '')[/^token (.*)/, 1]
```

**Review by lagr — APPROVED** [review]

File: lib/compeon/rack_tools/pipes/parse_token_with_key.rb (hunk 2 above)
Comment (timoschilling) [reply lib/compeon/rack_tools/pipes/parse_token_with_key.rb]:
```
token_string = request.env.fetch('HTTP_AUTHORIZATION', '')[/^token (?<token>.*)/, 'token']
```

Comment (lagr) [reply lib/compeon/rack_tools/pipes/parse_token_with_key.rb]: @timoschilling I considered that, but it seems kind of obvious that the capture group is a token, given the name of the variable and the leading `^token`. I am  not sure whether we need the word three times in a row 😉.

Comment (timoschilling) [reply lib/compeon/rack_tools/pipes/parse_token_with_key.rb]: I'm also ok with @lagr solution 😉

File: compeon-rack_tools.gemspec (hunk 1 above)
Comment (timoschilling) [inline compeon-rack_tools.gemspec]: I think this should only be a optional dependency

**Review by lagr — APPROVED** [review]

**Pull request merged**